### PR TITLE
ru-RU translation

### PR DESCRIPTION
**Pull request by ra30r**
Russian translation

### CODE REVIEW

**Comment by ra30r** [comment]
** I've run start command to double check md syntax, but only pushed source files.

File: docs/chapters/control/content.ru-RU.md [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+Желая изменить кривую, нам нужно изменить "вес"-а (* коофициенты) для каждой точки, по сути меняя интерполяцию. Имплементация оного предельно логична: нужно просто перемножить каждую точку на значение, меняющее ее 'силу'. Эти значения, по конвенции называют "весом" и мы можем учесть их запись в нашей оригинальной функции Безье:
+
+\[
+  Bézier(n,t) = \sum_{i=0}^{n}
+                \underset{binomial~term}{\underbrace{\binom{n}{i}}}
+                \cdot\
+                \underset{polynomial~term}{\underbrace{(1-t)^{n-i} \cdot t^{i}}}
+                \cdot\
+                \underset{weight}{\underbrace{w_i}}
+\]
```
Comment (Pomax) [inline docs/chapters/control/content.ru-RU.md:30]: the `binomial~term`, `polynomial~term`, and `weight` under the undersets can also be translated (the `~` is a special LaTeX space in maths mode)

File: docs/chapters/control/content.ru-RU.md [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+\[
+\left \{ \begin{matrix}
+  x = DARKRED[110] \cdot (1-t)^3 + DARKGREEN[25] \cdot 3 \cdot (1-t)^2 \cdot t + DARKBLUE[210] \cdot 3 \cdot (1-t) \cdot t^2 + AMBER[210] \cdot t^3 \\
+  y = DARKRED[150] \cdot (1-t)^3 + DARKGREEN[190] \cdot 3 \cdot (1-t)^2 \cdot t + DARKBLUE[250] \cdot 3 \cdot (1-t) \cdot t^2 + AMBER[30] \cdot t^3
+\end{matrix} \right.
+\]
+
+Это производит кривую, график которой мы видели в первой главе этой статьи.
+
+<graphics-element title="Our cubic Bézier curve" src="../introduction/cubic.js"></graphics-element>
```
Comment (Pomax) [inline docs/chapters/control/content.ru-RU.md:43]: this title can be translated

**Review by Pomax — CHANGES_REQUESTED** [review]
thank you! there's a few words that can still be translated

File: docs/chapters/explanation/content.ru-RU.md [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+  x = \cos(t) \\
+  y = \sin(t)
+\end{matrix} \right.
+\]
+
+Так-то. Координаты <i>x</i>/<i>y</i>. Связанные мистическим значением <i>t</i>.
+
+Таким образом, мы видим, что параметрические функции не определяют значение <i>y</i> через значение <i>x</i>, как обычные ф-ции, вместо этого они выводят и <i>x</i> и <i>y</i> из мистического значения <i>t</i>. Значит, для каждого значения <i>t</i>, существуют два соответствующих значения <i>x</i> и <i>y</i>, которые мы можем использовать для зарисовки кривой на графике. К примеру, приведенная выше параметрическая функция выводит координаты точки на круге: можно подставить любое значение <i>t</i> (от негативной до позитивной бесконечности), и на выводе мы всегда получим координаты точки на круге с центром в (0,0) и радиусом в 1. Зарисовав точки для <i>t</i> от 0 до 5, получаем следующее:
+
+<graphics-element title="A (partial) circle: x=sin(t), y=cos(t)" src="./circle.js">
```
Comment (Pomax) [inline docs/chapters/explanation/content.ru-RU.md:44]: (title)

File: docs/chapters/explanation/content.ru-RU.md [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+Обратите внимание, что 2 это то же, что 1+1, и 3 это 2+1 и 1+2 и 6 это 3+3... Как вы можете видеть, каждый раз поднимаясь на одно разрешение выше, мы начинаем и заканчиваем с единицы, а все промежуточные значения равны суме двух значений над ними в предыдущей строке. Так получаем последовательность цифр, называемой [треугольником Паскаля](https://ru.wikipedia.org/wiki/%D0%A2%D1%80%D0%B5%D1%83%D0%B3%D0%BE%D0%BB%D1%8C%D0%BD%D0%B8%D0%BA_%D0%9F%D0%B0%D1%81%D0%BA%D0%B0%D0%BB%D1%8F) (*в оригинале [другая ссылка](https://en.wikipedia.org/wiki/Pascal%27s_triangle) ).
+
+Так-же существует простой способ для выяснения подоплотной работы полиноминальных терминов: если мы переименуем <i>(1-t)</i> в <i>a</i> и <i>t</i> в <i>b</i>, убрав "веса", получим следующее:
+
+\[
+\begin{aligned}
+  linear &= BLUE[a] + RED[b] \\
+  square &= BLUE[a] \cdot BLUE[a] + BLUE[a] \cdot RED[b] + RED[b] \cdot RED[b] \\
+  cubic &= BLUE[a] \cdot BLUE[a] \cdot BLUE[a] + BLUE[a] \cdot BLUE[a] \cdot RED[b] + BLUE[a] \cdot RED[b] \cdot RED[b] + RED[b] \cdot RED[b] \cdot RED[b]\\
```
Comment (Pomax) [inline docs/chapters/explanation/content.ru-RU.md:87]: "linear", "square", and "cubic" can be translated

File: docs/chapters/explanation/content.ru-RU.md [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+\end{aligned}
+\]
+
+В целом это просто сума "каждого сочетания <i>a</i> и <i>b</i>", получаемая прогрессивной заменой <i>a</i> на <i>b</i> по ходу уравнения. Потому, это так-же довольно просто. Итак теперь вы знаете что такое биноминальные полиномы. Для полноты картины, ниже привожу их общую функцию:
+
+\[
+  Bézier(n,t) = \sum_{i=0}^{n}
+                \underset{binomial~term}{\underbrace{\binom{n}{i}}}
+                \cdot\
+                \underset{polynomial~term}{\underbrace{(1-t)^{n-i} \cdot t^{i}}}
```
Comment (Pomax) [inline docs/chapters/explanation/content.ru-RU.md:97]: `binomial~term` and `polynomial~term` can be translated

**Comment by Pomax** [comment]
For the conflicting files, you'll probably want to start working in a separate branch, instead of in `master`, so that you rebase your branch to master when necessary.

### making your code its own branch

You can turn the code that you have right now into a new branch called `translation`, without changing anything about the files, by using `git checkout -b translation`. You can now switch between which branch you want to work on by using `git checkout master`, to load the master branch, and `git checkout translation` to load the translation branch.

(Note that `checkout` means "checking out from your local copy of the repository files", not "checking out the code from github")

This means your master branch is now "free" to stay in sync with the one on this repository

### syncing your master branch

If you run `git checkout master` (after making sure you commit any code changes you might have of course!) you can sync it to the official master branch by first making sure you have the official remote available:

You can see a list of remotes by running `git remote`, which will probably show you this:

```bash
> git remote -v
origin  https://github.com/ra30r/BezierInfo-2 (fetch)
origin  https://github.com/ra30r/BezierInfo-2 (push)

```

this means git only knows about your own repo. To add this one as well, you can say:

```bash
> git remote add official https://github.com/Pomax/BezierInfo-2 
```

after this it should show the following:

```bash
> git remote -v
official https://github.com/Pomax/BezierInfo-2 (fetch)
official https://github.com/Pomax/BezierInfo-2 (push)
origin  https://github.com/ra30r/BezierInfo-2 (fetch)
origin  https://github.com/ra30r/BezierInfo-2 (push)

```

Which means any time you want to sync, you can now do:

```bash
> git checkout master
> git reset --hard
> git pull official master
```

That is, first we check out master, to make sure we have the right code loaded, then we reset it, so that we are certain there are no file changes, and then we pull the official branch state, which merges any commits not already merged in, in, so at the end of this process your `master` branch would be caught up.

The very very first time you do this, however, you can force-sync the master branch by "rebuilding" it from the official repo:

`> git checkout -B master official/master`

This throws away your local `master` branch, and then creates a new one with the same name, from the `official/master` branch instead. This is a useful command if you ever end up in a situation where you've been working in the master branch without realising it and you need to get the master branch back to the official state.

Also, should you ever need to remove a remote the command for that is `git remote rm nameofremote`.

### rebasing your working branch(es) so their PR doesn't conflict

In order to prevent PRs from having conflicts, you can rebase your branches so that they "start with" the current master branch, rather than whatever your local master branch was at the time you started your branch.

When you're on your not-master branch, first make sure any changes are captured in a commit (`git status` to verify there's nothing, but if there is, `git add ...` and `git commit -m "..."` to form a commit). Then, ensure your master branch is up to date (`git checkout master`, `git reset --hard`, `git pull official master`). Then the actual rebasing part:

```bash
> git checkout yourbranch
> git rebase -i master
```
this starts an interactive rebasing, where you get to say which commits you want to keep ("pick"), which commits should get merged into the previous commit instead of staying their own commit ("squash"), and which commits you don't even want to show up anymore (by deleting them entirely).

Now, unless there's a good reason to, just leave everything as is, and close the text editor that git opened for you. If it's a normal editor, easy enough, but if git opened `vi` it helps to know that you need to hit [esc] to tell it that you want the command prompt (which will show as `:` in the lower left) where you can then write `wq` and hit enter ("w"rite file, and then "q"uit).

https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request is a pretty decent guide that goes into all the details here.

once rebased, you can then push your now "properly synced branch" to github using the push mechanism:

```bash
> git push origin translation -f
```

where the only thing different is that `-f` flag, which means "force this push" because rebasing will have changes the commit history, and so git needs to be explicitly told that this is what you want.

**Pull request closed without merging**